srv/modules/runners/changed.py

```diff
@@ -0,0 +1,158 @@
+# -*- coding: utf-8 -*-
+# vim: tabstop=8 expandtab shiftwidth=4 softtabstop=4
+
+import os.path
+import hashlib
+import logging
+import salt.client
+
+"""
+This runner is here to detect config changes in the
+various configuration files to control service restarts.
+"""
+
+log = logging.getLogger(__name__)
+
+class Config(object):
+    def __init__(self, service_name):
+        self.base_dir = '/srv/salt/ceph/configuration/files/'
+        self.conf_dir = '/srv/salt/ceph/configuration/files/ceph.conf.d/'
+        self.checksum_dir = '/srv/salt/ceph/configuration/files/ceph.conf.checksum/'
+        self.checksum_file = self.checksum_dir + service_name + '.conf'
+        self.service_name = service_name
+        self.service_conf_files = [self.conf_dir + service_name + '.conf']
+        self.dependencies = self.dependencies()
+        self.rgw_configurations()
+        log.debug("dependencies: {}".format(self.dependencies))
+
+    def dependencies(self):
+        """
+        Services might depend on each other and have to trigger a restart.
+        I.e. Changes in the MDS section might affect the NFS-Ganesha service
+        hence the service should also be restarted.
+        # TODO for other services and complete tree
+        """
+        return { 'mds': ['mds'],
+                 'mon': ['mon'],
+                 'osd': ['osd'],
+                 'client': ['client'],
+                 'global': ['global'],
+               }
+
+    def rgw_configurations(self):
+        """
+        RadosGW allows custom configurations.  Include these roles with a
+        dependency on the global.conf.  Default to 'rgw' if not set.
+        """
+        local = salt.client.LocalClient()
+        roles = []
+        try:
+            roles = local.cmd("I@roles:master", 'pillar.get', [ 'rgw_configurations' ], expr_form="compound").values()[0]
```

Comment:
won't the `__pillar__` dunder dictionary work here?

Reply:
Not on my end :/

Reply:
ack